### PR TITLE
Fix issue #1: client may not be handling percent escape the right way

### PR DESCRIPTION
**Pull request by JJLionHeart**

### CODE REVIEW

File: urls.go [hunk 1]
```diff
@@ -266,7 +267,17 @@ func parseHost(hostish string) (host string, err error) {
 	if iphost := parseIPAddress(host); iphost != "" {
 		host = iphost
 	} else {
-		host = strings.ToLower(host)
+		host, _ = recursiveUnescape(host)
+		var temp_host bytes.Buffer
+		for _, c := range []byte(host) {
+			if (c >= 0x41 && c <=0x5A) || (c >= 0x61 && c <= 0x7A) {
```
Comment (lucamoller) [inline urls.go]: Add a comment explaining what were doing here? (e.g. Aplying ToLower only on ASCII characters)

Comment (JJLionHeart) [reply urls.go]: Done

**Comment by JJLionHeart** [comment]
PTAL

**Comment by lucamoller** [comment]
LGTM

**Pull request merged**